rpcs3/rpcs3qt/user_manager_dialog.cpp

```diff
@@ -0,0 +1,277 @@
+	QPushButton* push_create_user = new QPushButton(tr("Create User"), this);
+	QPushButton* push_login_user = new QPushButton(tr("Log In User"), this);
+
+	QPushButton* push_close = new QPushButton(tr("&Close"), this);
+	push_close->setAutoDefault(true);
+
+	// Button Layout
+	QHBoxLayout* hbox_buttons = new QHBoxLayout();
+	hbox_buttons->addWidget(push_remove_user);
+	hbox_buttons->addWidget(push_create_user);
+	hbox_buttons->addWidget(push_login_user);
+	hbox_buttons->addStretch();
+	hbox_buttons->addWidget(push_close);
+
+	// main layout
+	QVBoxLayout* vbox_main = new QVBoxLayout();
+	vbox_main->setAlignment(Qt::AlignCenter);
+	vbox_main->addWidget(m_table);
+	vbox_main->addLayout(hbox_buttons);
+	setLayout(vbox_main);
+
+	UpdateTable();
+
+	QByteArray geometry = m_gui_settings->GetValue(gui::sd_geometry).toByteArray();
+	if (geometry.isEmpty() == false)
+	{
+		restoreGeometry(geometry);
+	}
+
+	// Connects and events
+	connect(push_close, &QAbstractButton::clicked, this, &user_manager_dialog::close);
+	connect(push_remove_user, &QAbstractButton::clicked, this, &user_manager_dialog::OnUserRemove);
+	connect(push_create_user, &QAbstractButton::clicked, this, &user_manager_dialog::OnUserCreate);
+	connect(push_login_user, &QAbstractButton::clicked, this, &user_manager_dialog::OnUserLogin);
+	connect(m_table, &QTableWidget::itemDoubleClicked, this, &user_manager_dialog::OnUserLogin);
+	connect(m_table->horizontalHeader(), &QHeaderView::sectionClicked, this, &user_manager_dialog::OnSort);
+	connect(m_table, &QTableWidget::customContextMenuRequested, this, &user_manager_dialog::ShowContextMenu);
+	connect(m_table, &QTableWidget::cellChanged, [&](int row, int col)
+	{
+		int originalIndex = m_table->item(row, 0)->data(Qt::UserRole).toInt();
+		UserAccount* originalEntry = m_user_list[originalIndex];
+		QString originalDirName = qstr(originalEntry->GetUserDir());
+	});
+}
+
+void user_manager_dialog::UpdateTable()
+{
+	if (m_dir == "")
+	{
+		// fmt::format(%shome ... is harder to read than straight concatenation.
+		m_dir = Emu.GetHddDir() + "home";
+	}
+
+	// Get the user folders in the home directory.
+	m_user_list = GetUserAccounts(m_dir);
+
+	// Clear and then repopulate the table with the list gathered above.
+	m_table->clearContents();
+	m_table->setRowCount(static_cast<int>(m_user_list.size()));
+
+	int row = 0;
+	for (UserAccount* user : m_user_list)
+	{
+		// For indicating logged-in user.
+		QFont boldFont;
+		boldFont.setBold(true);
+
+		QString userId = qstr(user->GetUserId());
+		QString userName = qstr(user->GetUserName());
+
+		QTableWidgetItem* userIdItem = new QTableWidgetItem(userId);
+		userIdItem->setData(Qt::UserRole, row); // For sorting to work properly
+		userIdItem->setFlags(userIdItem->flags() & ~Qt::ItemIsEditable);
+
+		// Compare current config value with the one in this user (only 8 digits in userId)
+		if (m_emu_settings->GetSetting(emu_settings::SelectedUser)
```

Comment:
take this out of the loop... and why this weird two liner

Reply:
also: you are using the exact same code a few lines below. just set both bold at the end of the loop